lib/bson/regexp.rb

```diff
@@ -154,6 +154,12 @@ def compile
       #
       # @since 3.0.0
       def initialize(pattern, options = '')
+        if pattern.include? NULL_BYTE
+          raise Error::ExtJSONParseError, "Regex pattern cannot contian a null byte: #{key}"
```

Comment:
The exception class here and on the line below should be `ArgumentError` or  a new class added under `BSON::Error`. The extended json parse error isn't applicable here because this method isn't parsing extended json.

Reply:
I think "regexp" will also be more appropriate than "regex" here since the input is Ruby variables and the Ruby class is called "regexp".